core/lib/zksync_core/src/metadata_calculator/recovery/mod.rs

```diff
@@ -205,8 +207,16 @@ impl GenericAsyncTree {
             concurrency_limit: recovery_pool.max_size() as usize,
             events: Box::new(RecoveryHealthUpdater::new(health_updater)),
         };
-        tree.recover(snapshot, recovery_options, &recovery_pool, stop_receiver)
-            .await
+        let tree = tree
+            .recover(snapshot, recovery_options, &recovery_pool, stop_receiver)
+            .await?;
+        if tree.is_some() {
+            // Only report latency if recovery wasn't canceled
```

Comment:
nit:
```suggestion
            // Only report latency if recovery wasn't cancelled
```

Reply:
Ah, sorry apparently it's one of those British vs US spelling thingies

Reply:
"cancelled" is reported as incorrect by `zk spellcheck` 🙃